labgrid/driver/sshdriver.py

```diff
@@ -392,9 +393,12 @@ def get_status(self):
     @Driver.check_active
     @step(args=['filename', 'remotepath'])
     def put(self, filename, remotepath=''):
+        ssh_prefix = self.ssh_prefix
+        if self.sftp:
+            ssh_prefix.append("-s")
         transfer_cmd = [
             "scp",
-            *self.ssh_prefix,
+            *ssh_prefix,
```

Comment:
If I read this correctly, you are passing `-s` to scp. Looking at `man scp`, I don't see such an option. I guess you're referring to ssh's `-s` here? No idea how this should work, though. How exactly did you test this?

Reply:
I just checked on my up-to-date Arch machine and `-s` indeed does not exist for `scp` here. However, on my other machine `-s` does exist. According the the release notes from OpenSSH, `-s` is removed in version 9 (https://www.openssh.com/releasenotes.html#9.0). So this will be fine on targets with OpenSSH < 9. However, it will probably not work on targets with OpenSSH > 9, since `scp` uses the `sftp` protocol by default here.

Sorry I missed this detail. What do you think the best solution is to handle both protocols?

Reply:
> I just checked on my up-to-date Arch machine and `-s` indeed does not exist for `scp` here. However, on my other machine `-s` does exist. According the the release notes from OpenSSH, `-s` is removed in version 9 (https://www.openssh.com/releasenotes.html#9.0). So this will be fine on targets with OpenSSH < 9. However, it will probably not work on targets with OpenSSH > 9, since `scp` uses the `sftp` protocol by default here.

Are you sure about this? I tested `scp` from Debian's `openssh-client` package in versions 8.4 and 9.0. Neither have the `-s` option.

> Sorry I missed this detail. What do you think the best solution is to handle both protocols?

`-o SessionType=subsystem` should work, right?

Reply:
> > I just checked on my up-to-date Arch machine and `-s` indeed does not exist for `scp` here. However, on my other machine `-s` does exist. According the the release notes from OpenSSH, `-s` is removed in version 9 (https://www.openssh.com/releasenotes.html#9.0). So this will be fine on targets with OpenSSH < 9. However, it will probably not work on targets with OpenSSH > 9, since `scp` uses the `sftp` protocol by default here.
> 
> Are you sure about this? I tested `scp` from Debian's `openssh-client` package in versions 8.4 and 9.0. Neither have the `-s` option.

Oh, that's not true. scp from `openssh-client` 9.0 has it, but the man page seems outdated. We should still stick to:

> > Sorry I missed this detail. What do you think the best solution is to handle both protocols?
> 
> `-o SessionType=subsystem` should work, right?

Reply:
> I guess this will still be around after 2022..

Fingers crossed!

I tried substituting `-s` for `-o SessionType=subsystem`,  but it doesn't work even though the documentation states that it should be the same. I will troubleshoot more tomorrow.

Reply:
Could you push your WIP state so I can have a look?

Reply:
Done

Reply:
No luck here either:
```
$ scp -v -F /dev/null -o SessionType=subsystem somefile root@target:/tmp/
subsystem request failed on channel 0
lost connection
```
On the target sshd side:
```
subsystem request for scp -v -t /tmp/ by user root failed, subsystem not found
```

So `-s` and `-o SessionType=subsystem` do not seem to be interchangeable.

In this case I'd opt for passing `-s` and rename the SSHDriver's `sftp` attribute to something like `explicit_sftp_mode`. We should document that this option should only by passed for OpenSSH version >=8.6, <9 and that the scp flag will be removed in 2022.

Reply:
I agree with this approach and I just pushed some new changes.

Reply:
@Bastian-Krause Did you have a chance to look at these changes and [1009](https://github.com/labgrid-project/labgrid/pull/1009) yet?